client/js/chart/index.js

```diff
@@ -17,16 +17,87 @@ export function createChart(el, rankingHistory, rankingDates) {
   seriesContainer.setAttribute('class', 'series-container');
   svg.append(seriesContainer);
 
-  let index = 0;
-  const colors = ['blue', 'red', 'green', 'purple', 'orange'];
-
-  // get only the first 7 tracks for now
-  for (const history of [...rankingHistory.values()].slice(0, 7)) {
-    const color = colors[index % colors.length];
-    const series = createSeries(color, history);
+  /** @type {SVGGElement[]} */
+  const seriesElements = [];
+  const rankingHistoryEntries = [...rankingHistory];
+
+  rankingHistoryEntries.forEach(([, history], index) => {
+    const hue = index * 15 % 360;
+    const color = `hsl(${hue},50%,50%)`;
+    const series = createSeries(svg, color, history);
     seriesContainer.append(series);
-    index++;
-  }
+    seriesElements.push(series);
+  });
+
+  let activeSeries = null;
+  let activeX = null;
+  let activeY = null;
+
+  // do some hit-testing
+  svg.addEventListener('mousemove', (ev) => {
+    const svgBounds = svg.getBoundingClientRect();
+    const chartX = ev.clientX - svgBounds.x;
+    const chartY = ev.clientY - svgBounds.y;
+
+    // index in history
+    const x = Math.round(chartX / RUN_SCALE_X);
+    // value in history
+    const y = Math.round(chartY / RUN_SCALE_Y);
+
+    // don't recalculate the hit if we're in the same place
+    // as the last hit
+    if (activeX === x && activeY === y) return;
+
+    let hit = null;
+
+    rankingHistoryEntries.forEach(([, history], index) => {
+      // include only series where there is a data point for this x value
+      if (x >= history.length || history[x] === null) return;
+
+      // get distance from mouse y to series y for each series
+      const dist = history[x] - y;
+
+      // filter items where distance is too great; there should only be one
+      // remaining (impossible to be less than 0.4 away from both 1 and 2, for
+      // example)
+      if (Math.abs(dist) > 0.4) return;
+
+      hit = index;
+    });
+
+    if (activeSeries && activeSeries !== hit) {
+      const seriesElement = seriesElements[activeSeries];
+      seriesElement.dispatchEvent(new CustomEvent('series-clear'));
+    }
```

Comment:
what is `activeSeries`?? Is this condition trying to see whether `activerSeries` exists and has not been hit yet?

Reply:
It is the series that the user is currently hovering over. How is `currentSeries` a more obvious name than `activeSeries`?